Sources/_StringProcessing/Engine/Processor.swift

```diff
@@ -508,15 +610,18 @@ extension Processor {
       controller.step()
 
     case .assertBy:
-      let reg = payload.assertion
-      let assertion = registers[reg]
+      let (kind,
+           anchorsMatchNewlines,
+           usesSimpleUnicodeBoundaries,
+           usesASCIIWord,
+           semanticLevel) = payload.assertion
       do {
-        guard try assertion(
-          &wordIndexCache,
-          &wordIndexMaxIndex,
-          input,
-          currentPosition,
-          subjectBounds
+        guard try regexAssert(
+          by: kind,
+          anchorsMatchNewlines,
+          usesSimpleUnicodeBoundaries,
+          usesASCIIWord,
+          semanticLevel
```

Comment:
When we have the struct that just wraps the `UInt64` and has API for these queries, there's no need to pack/unpack it when evaluating. We can just branch directly on the bits.